contracts/aFloorMigration.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.7.5;
+
+import "./interfaces/IERC20.sol";
+import "./libraries/SafeMath.sol";
+import "./types/Ownable.sol";
+
+contract AlphaFloorMigration is Ownable {
+    using SafeMath for uint256;
+
+    uint256 public swapEndBlock;
+
+    IERC20 public FLOOR;
+    IERC20 public aFLOOR;
+    
+    bool public isInitialized;
+
+    modifier onlyInitialized() {
+        require(isInitialized, "not initialized");
+        _;
+    }
+
+    modifier notInitialized() {
+        require(!isInitialized, "already initialized" );
+        _;
+    }
+
+    constructor() {}
+
+    function initialize (
+        address _FLOOR,
+        address _aFLOOR,
+        uint256 _swapDuration
+    ) public notInitialized() onlyOwner() {
+        require(_swapDuration < 2_000_000, "swap duration too long");
+        FLOOR = IERC20(_FLOOR);
```

Comment:
Might be a good idea to check for zero addresses since there's no changing the variables after they're set.